src/Schema/Comparator.php

```diff
@@ -432,95 +434,76 @@ public function diffForeignKey(ForeignKeyConstraint $key1, ForeignKeyConstraint
-                $changedProperties[] = 'scale';
+    /**
+     * Detects if a changed property has an actual effect on the used platform
+     */
+    private function changedPropertyHasEffect(
+        string $property,
+        Column $column1,
+        Column $column2,
+        AbstractPlatform $platform
+    ): bool {
+        $column2WithoutChange = clone $column2;
+
+        if ($column1->hasCustomSchemaOption($property) || $column2->hasCustomSchemaOption($property)) {
+            if (! $column1->hasCustomSchemaOption($property)) {
+                $options = $column2WithoutChange->getCustomSchemaOptions();
+                unset($options[$property]);
+                $column2WithoutChange->setCustomSchemaOptions($options);
+            } else {
+                $column2WithoutChange->setCustomSchemaOption($property, $column1->getCustomSchemaOption($property));
             }
-        }
-
-        // A null value and an empty string are actually equal for a comment so they should not trigger a change.
-        if (
-            $properties1['comment'] !== $properties2['comment'] &&
-            ! ($properties1['comment'] === null && $properties2['comment'] === '') &&
-            ! ($properties2['comment'] === null && $properties1['comment'] === '')
-        ) {
-            $changedProperties[] = 'comment';
-        }
-
-        $customOptions1 = $column1->getCustomSchemaOptions();
-        $customOptions2 = $column2->getCustomSchemaOptions();
-
-        foreach (array_merge(array_keys($customOptions1), array_keys($customOptions2)) as $key) {
-            if (! array_key_exists($key, $properties1) || ! array_key_exists($key, $properties2)) {
-                $changedProperties[] = $key;
-            } elseif ($properties1[$key] !== $properties2[$key]) {
-                $changedProperties[] = $key;
+        } elseif ($column1->hasPlatformOption($property) || $column2->hasPlatformOption($property)) {
+            if (! $column1->hasPlatformOption($property)) {
+                $options = $column2WithoutChange->getPlatformOptions();
+                unset($options[$property]);
+                $column2WithoutChange->setPlatformOptions($options);
+            } else {
+                $column2WithoutChange->setPlatformOption($property, $column1->getPlatformOption($property));
             }
+        } else {
+            $column2WithoutChange->{'set' . $property}($column1->{'get' . $property}());
         }
 
-        $platformOptions1 = $column1->getPlatformOptions();
-        $platformOptions2 = $column2->getPlatformOptions();
-
-        foreach (array_keys(array_intersect_key($platformOptions1, $platformOptions2)) as $key) {
-            if ($properties1[$key] === $properties2[$key]) {
-                continue;
-            }
-
-            $changedProperties[] = $key;
-        }
+        $sqlWithChange    = $platform->getCreateTableSQL(new Table('compare_table', [$column2]));
```

Comment:
This looks like overkill. We shouldn't have to create a table to compare columns. If there's no platform API for generating a column SQL, it should be introduced.

Reply:
Done in 5dbb71531556225f189f5bc1cd26009b762d0c84.